xen/include/public/io/sndif.h

```diff
@@ -246,17 +307,77 @@
 #define XENSND_OP_WRITE                 3
 #define XENSND_OP_SET_VOLUME            4
 #define XENSND_OP_GET_VOLUME            5
+#define XENSND_OP_MUTE                  6
+#define XENSND_OP_UNMUTE                7
 
 /*
  * The maximum amount of shared pages which can be used in any request
  * from the frontend driver to the backend driver
+ * Pages must be of XC_PAGE_SIZE length.
  */
 #define XENSND_MAX_PAGES_PER_REQUEST    10
```

Comment:
Will definitely limit number of channels to something more realistic, say 16 or 32